### DISCUSSION TITLE
Persistent modification of nginx.conf

### DISCUSSION THREAD

**Discussion by tangothr33** [telekom-security/tpotce#1219] [Q&A]
Hello!

I am trying to generate a CSV report and query the Kibana interface by using its API. For automation, this happens from a Python script. The request fails because my URI is too long.
Given the fact that the request must contain the parameters, my other option is to modify the nginx.conf with the line of "large_client_header_buffers  1 1k;" until the point I no longer receive the error.
Since I'm not that familiar with Docker, I'm wondering what would be the safest way to implement this change. 

Any advice or idea is welcome, thank you!

**Comment by eogradney** [comment]
Hello!

I have several persistent modifications in the nginx configuration.   Here is how I did it: 

     1. Stopped tpot services.
     2. Set the nginx container to mount in read-write mode.
          - change read_only = true to read_only = false in tpot.yml
     3. Started tpot services.
     4. Updated the nginx config file(s) on the running container.  
     5. Committed the changes to a new nginx container image. 
         - docker commit <running nginx docker ID> <new container image name:tag>
     6. Stopped tpot services.
     8. Updated tpot.yml with the new container image name. 
       *you can also commit changes and keep the original image name and not have to change tpot.yml*
     9. Restarted tpot services

Once you do this, you'll have to manage the nginx container yourself.  The default cron job that replaces all the containers nightly needs to be disabled or modified, so your changes are not overwritten every 24 hours.  I disabled the job completely and run updates manually as needed.  I keep the nginx container files in my own github repo.

Hope this helps.
